src/parsers/manifest/dash/parse_adaptation_sets.ts

```diff
@@ -334,19 +343,11 @@ export default function parseAdaptationSets(
         isDub = true;
       }
 
-      const isClosedCaption = type === "text" &&
-                              accessibility != null &&
-                              isHardOfHearing(accessibility) ? true :
-                                                               undefined;
-      const isAudioDescription = type === "audio" &&
-                                 accessibility != null &&
-                                 isVisuallyImpaired(accessibility) ? true :
-                                                                     undefined;
-
-      const isSignInterpreted = type === "video" &&
-                                accessibility != null &&
-                                hasSignLanguageInterpretation(accessibility) ? true :
-                                                                               undefined;
+      const isClosedCaption = type === "text" && accessibilities?.some(isHardOfHearing);
```

Comment:
I’ve noticed here that you changed the possible values for `isClosedCaption`.  If the `type` is not equal to « text », `isClosedCaption` will be `false`. Actually, it seems to be logical, and it seems to be a good change.

 However, the `isClosedCaption` can still be equal to undefined if `accessibilities` is undefined. I think that it is not really clear when using optional chaining (`true ?? undefined = undefined`, so `true ?? someObject?.attribute = undefined` if `someObject` is `undefined`). 
 
 Could you please write it differently, while keeping this logic, so it can be clearer?

Reply:
> true ?? undefined === undefined, so true ?? someObject?.attribute === undefined

I think you mixed-up `??` and `&&` in your sentence.
It's
```true && undefined === undefined, so true && someObject?.attribute === undefined```

Reply:
I'm not quite sure what you want here, do you want it to return `undefined` in the case where `accessibilities` is `undefined` but just without using optional chaining? Or do you want it to return `false` in that case?

Reply:
> do you want it to return undefined in the case where accessibilities is undefined but just without using optional chaining?

Yes, in my opinion, that would be ideal!

Reply:
Something like:
```
const isClosedCaption = type === "text" && accessibilities && accessibilities.some(isHardOfHearing);
```

or in order to explicitly return undefined:

```
function hasAccessibilityOfType(accessibilities, predicate) {
    if (!accessibilities) {
       return undefined;
    }

    return accessibilities.some(predicate);
}

const isClosedCaption = type === "text" && hasAccessibilityOfType(accessibilities, isHardOfHearing);
```

Which one do you prefer or do you want it done in some other way?

Reply:
Well, we are used to only use the `&&` operator with booleans. We’ve decided this to improve readability and to avoid possible errors.

I consider that the `true && someObj?.attr` is confusing as it uses the `&&` operator with an `undefined` and it is not clear that `someObj?.attr` may be `undefined`.  What do you think of getting rid of the operator?

It probably implies using a `let`, but it is not a problem for us to use `let` instead of `const` in these situations, as we prefer the code to be clear.

Reply:
Aha, I see! I thought that you just didn't want the optional chaining operator - but I didn't catch that you didn't wanted the `&&` operator with a potential undefined as well, that's why I didn't consider that in the suggestions.

So, is it something like this you're after, I'm fine with that as well:
```
let isClosedCaption;
if (type === "text" && accessibilities !== undefined) {
    isClosedCaption = accessibilities.some(isHardOfHearing)
}
```

If not, I think I must ask for an example.

Reply:
Actually, in order to reproduce the logic of : 
`const isClosedCaption = type === "text" && accessibilities?.some(isHardOfHearing);`

I would rather see something as : 
```
let isClosedCaption;
if (type !== "text") {
  isClosedCaption = false;
} else if (accessibilities !== undefined) {
  isClosedCaption = accessibilities.some(isHardOfHearing);
}
```
If the type is not "text", it is explicitly not a closed caption. If there are accessibilities, the result depends on it. If there are no accessibilities, then it is not defined.

Reply:
Okay, I'll change to that version for all cases!